pkg/cluster/k8sres.go

```diff
@@ -939,6 +939,10 @@ func deduplicateEnvVars(input []v1.EnvVar, containerName string, logger *logrus.
 func (c *Cluster) getPodEnvironmentConfigMapVariables() ([]v1.EnvVar, error) {
 	configMapPodEnvVarsList := make([]v1.EnvVar, 0)
 
+	if len(c.Spec.Env) > 0 {
+		configMapPodEnvVarsList = append(configMapPodEnvVarsList, c.Spec.Env...)
+	}
```

Comment:
Lets to do this only once in `generateStatefulSet`. First get c.Spec.Env because it takes precedence over all other global configs. With the result from `getPodEnvironmentConfigMapVariables` and `getPodEnvironmentSecretVariables` we can do the `isEnvVarPresent` in  `generateStatefulSet`, too. This way, both functions will just do what their names tell us they are supposed to do.

Reply:
Already did it, what do you think about `generateSpiloPodEnvVars` location?

Reply:
Yes, also a good idea, to move just everything from `generateStatefulSet` to `generateSpiloPodEnvVars`